src/archive.rs

```diff
@@ -256,6 +257,7 @@ impl<'a, R: Read> Iterator for Entries<'a, R> {
     }
 }
 
+#[allow(unused_assignments)] // https://github.com/rust-lang/rust/issues/22630
```

Comment:
I think it would be best to remove this or move the comment to where the warning is printed instead.

Reply:
Done.